lib/Doctrine/DBAL/Connection.php

```diff
@@ -1954,4 +1957,60 @@ public function ping()
             return false;
         }
     }
+
+    /**
+     * @internal
+     *
+     * @param string                                           $sql
+     * @param array<int, mixed>|array<string, mixed>           $params
+     * @param array<int, int|string>|array<string, int|string> $types
+     *
+     * @throws DBALException
+     *
+     * @psalm-return never-return
+     */
+    public function handleExceptionDuringQuery(Throwable $e, $sql, array $params = [], array $types = []): void
```

Comment:
It's internal. Should we maybe also make it `final`?
```suggestion
    final public function handleExceptionDuringQuery(Throwable $e, $sql, array $params = [], array $types = []): void
```

Reply:
Make sense.

Reply:
Unfortunately, it won't work because of the existing test that now has to mock this method to pass:

https://github.com/doctrine/dbal/blob/0a5bac1c2ce1d1b68944541e6a74e2325d274847/tests/Doctrine/Tests/DBAL/StatementTest.php#L134-L136